### PR TITLE
Add devtest labs python CLI and two new actions: auth and createvm

### PR DESCRIPTION
**Pull request by ryanvog**
Add script to create a lab virtual machine (in a new environment) based on a series of command line arguments.

## General
```
python dtl.py --help
usage: dtl.py <principal> <tenant ID> <secret> <subscription ID> action

Provides command-line access to Azure DevTest Labs

optional arguments:
  -h, --help            show this help message and exit
  -v, --version         show program's version number and exit
  -sp PRINCIPAL, --principal PRINCIPAL
                        Your Azure Application/Client ID
  -p SECRET, --secret SECRET
                        Your Client secret.
  -t TENANT, --tenant TENANT
                        Your tenant ID. See https://msdn.microsoft.com/en-
                        us/library/azure/dn790557.aspx for more information.
  -s SUBSCRIPTION, --subscription SUBSCRIPTION
                        Your Azure subscription ID

Actions:
  {auth,labs,createvm}
    auth                Authorizes access to Azure resources
    labs                Provides access to individual labs
    createvm            Creates a lab virtual machine based on additional
                        parameters

```

## Auth
```
python dtl.py auth --help

usage: dtl.py <principal> <tenant ID> <secret> <subscription ID> action auth
       [-h] [-vb] [-q]

optional arguments:
  -h, --help      show this help message and exit
  -vb, --verbose  Log verbose output to stdout
  -q, --quiet     When set, suppresses tool messages from printing

```
## Labs
```
python dtl.py labs --help

usage: dtl.py <principal> <tenant ID> <secret> <subscription ID> action labs
       [-h] [-l LABNAME] [-vb] [-q]

optional arguments:
  -h, --help            show this help message and exit
  -l LABNAME, --labname LABNAME
                        The name of the lab
  -vb, --verbose        Log verbose output to stdout
  -q, --quiet           When set, suppresses tool messages from printing

```
## Createvm
```
usage: dtl.py <principal> <tenant ID> <secret> <subscription ID> action createvm
       [-h] -l LABNAME -n NAME -tn TEMPLATENAME -sz SIZE [-vu VMUSERNAME]
       [-vp VMPASSWORD] [-ssh VMSSHKEY] [-vb] [-q]

optional arguments:
  -h, --help            show this help message and exit
  -l LABNAME, --labname LABNAME
                        The name of the lab
  -n NAME, --name NAME  The name of the new virtual machine
  -tn TEMPLATENAME, --templatename TEMPLATENAME
                        The name of the base image template to use
  -sz SIZE, --size SIZE
                        The size of the new virtual machine
  -vu VMUSERNAME, --vmusername VMUSERNAME
                        The user name of the new virtual machine
  -vp VMPASSWORD, --vmpassword VMPASSWORD
                        The password of the new virtual machine
  -ssh VMSSHKEY, --vmsshkey VMSSHKEY
                        The SSH key of the new virtual machine
  -vb, --verbose        Log verbose output to stdout
  -q, --quiet           When set, suppresses tool messages from printing

```

### CODE REVIEW

File: Scripts/linux/basiclabtemplate.json [hunk 1]
```diff
@@ -0,0 +1,70 @@
+{
+  "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "vmName": {
+      "type": "string",
+      "metadata": {
+        "description": "The name of the new vm to be created."
```
Comment (drwill-ms) [inline Scripts/linux/basiclabtemplate.json]: capitalize VM

File: Scripts/linux/dtl.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+import argparse
+import dtlcommands
+import dtlproviders
+
+
+class DevTestLabsMain:
+    """The boostrap used to instantiate the DTL CLI
+
+    Attributes:
+        _argParser [private] (ArgumentParser) - The argument parser used to present the CLI interface and parse the command-line.
```
Comment (drwill-ms) [inline Scripts/linux/dtl.py]: CLI interface = command-line interface interface

File: Scripts/linux/dtlcommands.py [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+        """Constructs the command-line arguments used to support this command.
+
+        Args:
+            argParser (ArgumentParser) - the arguments parser used to parse the command-line.
+
+        Returns:
+            None
+        """
+        argParser.add_argument_group('Authentication')
+        argParser.add_argument('principal', help='Your Azure user name or Application/Client ID')
```
Comment (drwill-ms) [inline Scripts/linux/dtlcommands.py]: Sometimes you write "user name" and sometimes "username". Minor but would be nice if this were consistent.

File: Scripts/linux/dtlcommands.py [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+        """Constructs the command-line arguments used to support this command.
+
+        Args:
+            argParser (ArgumentParser) - the arguments parser used to parse the command-line.
+
+        Returns:
+            None
+        """
+
+        argParser.add_argument('-vb', '--verbose',
```
Comment (drwill-ms) [inline Scripts/linux/dtlcommands.py]: two arguments on same line

File: Scripts/linux/dtlcommands.py [hunk 5]
```diff
@@ -0,0 +1,359 @@
...
+        Returns:
+           0 if successful, 1 otherwise.
+        """
+
+        settings.vmTemplateFileName = self._vmTemplateFileName
+        labSvc = dtllabservice.LabService(settings)
+
+        return labSvc.createVm(settings.subscriptionID,
+                               self._vmTemplateFileName,
+                               settings.labname,
```
Comment (drwill-ms) [inline Scripts/linux/dtlcommands.py]: Curious why the settings properties are all lowercase

File: Scripts/linux/dtlcommands.py (hunk 4 above)
Comment (ryanvog) [inline Scripts/linux/dtlcommands.py]: This is a convention followed by the argparse lib.

**Comment by drwill-ms** [comment]
misspelled "suppresses" in the help text above with --quiet. is that from python?

**Comment by drwill-ms** [comment]
All I got is wording consistency feedback. Don't know enough about Python to suggest much else. Looks pretty cool.

File: Scripts/linux/dtlcommands.py (hunk 3 above)
Comment (ryanvog) [inline Scripts/linux/dtlcommands.py]: Fixed

File: Scripts/linux/basiclabtemplate.json (hunk 1 above)
Comment (ryanvog) [inline Scripts/linux/basiclabtemplate.json]: Fixed

**Pull request closed without merging**